tempesta_fw/http_parser.c

```diff
@@ -2251,6 +2312,116 @@ __req_parse_host(TfwHttpReq *req, unsigned char *data, size_t len)
 	return r;
 }
 
+/**
+ * Function for parsing 'Referer' header value. This is a partial
+ * copy from parsing procedure for request start line URI with some 
+ * simplifications according to RFC 7231 chapter 5.5.2.
+ */
+static int
+__req_parse_referer(TfwHttpMsg *hm, unsigned char *data, size_t len)
+{
+	int r = CSTR_NEQ;
+	unsigned char *off_ptr = data;
+	__FSM_DECLARE_VARS(hm);
+
+	__FSM_START(parser->_i_st) {
```

Comment:
Is it required to fully parse `Referer` header? We parse it for frang and ddos mitigation. How the attack work: attacker places  link to the victim on popular web site in such a way that every visitor send request to victim. It's expected that `referer` header is constructed correctly, just need to check that value is a part of URI alphabet. 

More important is how we are going to use the header in frang. It's seems that string match is the only way here. From this point of view, it's may be sufficient to start a new chunk at the beginning of uri and mark it with `TFW_STR_VALUE` flag.

comment from @krizhanovsky  is required here.

Reply:
Yes, there were several discussions in hacker forums (like antichat) how to disable Rerefer header for iFrame attacks and basically there is no good solution since Referer is the fundamental security header and browsers don't allow to mangle it. So there is no much sense to introduce the parsing overhead for the header which is safe for iFrame attack and hasn't much sense to be verified for other attacks.